proposals/stdlib/instant.md

```diff
@@ -0,0 +1,1365 @@
+  handling of `java.time.Instant` will automatically work.
+* A lot of APIs accept `java.time.Instant` as parameters:
+  <https://grep.app/search?q=%28Instant%20>.
+
+Disadvantages also exist:
+
+* As noted above, there are several differences between how methods in
+  `java.time.Instant` and `kotlinx.datetime.Instant` work.
+  Most notably, several APIs that have the same name behave differently:
+  `now` is deprecated, and `parse` does not accept the same set of strings.
+  All remaining methods that have different behaviors between Kotlin and Java
+  also have slight differences in names.
+  For `parse`, this means that `java.time.Instant.parse` and
+  `kotlin.time.Instant.parse` would be functions with the exact same interface
+  but different behavior.
+* Increased complexity of adding new API.
+
+Neutral points:
+
+* If Java adds some new API with a name we already provide but with a different
+  behavior, we will need to hide it.
+  However, there doesn't seem to be such API at the moment.
+* Undesirable interface implementations: `Temporal`, `TemporalAdjuster`.
+  However, we do not intend to implement these interfaces on
+  `kotlin.time.Instant`, so this is not a compatibility problem.
+* Mapping `kotlin.time.Instant` to `java.time.Instant` should not make
+  the migration easier, as `kotlinx-datetime` is a multiplatform
+  library.
+
+The technical details of how this mapping could be achieved is outside the scope
+of this document.
+As of writing, a KEEP dedicated to adding atomic values to the Standard Library
+is exploring this topic in great depth.
+
+#### `java.io.Serializable`
+
+We intend to add its support to `kotlin.time.Instant`, but the specific
+implementation depends on how we proceed with the migration.
+
+We do not have any hard constraints on the implementation.
+It is possible that `kotlinx-datetime` releases with its own implementation of
+`java.io.Serializable` for `kotlinx.datetime.Instant`, but these formats don't
+have to be compatible: whenever someone serializes `kotlinx.datetime.Instant`,
+they expect to receive just that on deserialization.
+When the code serializing `kotlinx.datetime.Instant` is migrated to
+`kotlin.time.Instant`, the old serialization implementation is no longer
+relevant.
+
+### Additional platform-specific converters
+
+On all platforms, there are entities fulfilling the same purpose as
+`Instant`, and for some of them, we provide automatic conversion functions:
+
+```kotlin
+// JVM
+
+public fun kotlin.time.Instant.toJavaInstant(): java.time.Instant
+
+public fun java.time.Instant.toKotlinInstant(): kotlin.time.Instant
+```
+
+```kotlin
+// JS
+
+/**
+ * Converts the [Instant] to an instance of JS [Date].
+ *
+ * The conversion is lossy: JS uses millisecond precision to represent dates,
+ * and [Instant] allows for nanosecond resolution.
+ */
+public fun Instant.toJSDate(): Date
+
+public fun Date.toKotlinInstant(): Instant
+```
+
+```kotlin
+// Darwin
```

Comment:
According to the UUID KEEP, the stdlib is not specialized for various native targets today. These extensions would no longer be possible.